File: objc/Strategy/Strategy/LaughAnalyzeStrategy.m
#import "LaughAnalyzeStrategy.h"

@implementation LaughAnalyzeStrategy

-(void)analyze:(NSArray<Post*>*)posts {
    int postCount = [posts count];
    int totalLaughCount = 0;
    for(Post* post in posts)
        totalLaughCount += post.laughCount;
    NSLog(@"Laugh Value: %lf", (double)totalLaughCount / postCount);
}

@end

File: swift/Strategy/Strategy/LaughAnalyzeStrategy.swift
import Foundation

class LaughAnalyzeStrategy : PostAnalyzeStrategy {
    func analyze(_ posts: [Post]) -> Void {
        let postCount: Int = posts.count
        var totalLaughCount: Int = 0
        for post in posts {
            totalLaughCount += post.laughCount
        }
        let val: Double = Double(totalLaughCount / postCount)
        print("Laugh Value: \(val)")
    }
}
